R/get.R

```diff
@@ -0,0 +1,597 @@
+#' }
+#' 
+#' @export
+get_usage_static <- function(src, content_guid = NULL, 
+                             min_data_version = NULL,
+                             from = NULL,
+                             to = NULL,
+                             limit = 20,
+                             previous = NULL,
+                             nxt = NULL,
+                             asc_order = TRUE){
+  validate_R6_class("Connect", src)
+  
+  res <- src$inst_content_visits(
+    content_guid = content_guid, 
+    min_data_version = min_data_version, 
+    from = from,
+    to = to, 
+    limit = limit, 
+    previous = previous, 
+    nxt = nxt,
+    asc_order = asc_order
+  )
+  
+  res <- page_cursor(src, res, limit = limit)
+  
+  out <- parse_connectapi(res)
+  
+  return(out)
+}
+
+
+#' Get Audit Logs from RStudio Connect Server
+#' 
+#' @param src The source object
+#' @param limit The number of records to return. 
+#' @param previous Retrieve the previous page of Shiny application usage 
+#' logs relative to the provided value. This value corresponds to an internal 
+#' reference within the server and should be sourced from the appropriate 
+#' attribute within the paging object of a previous response.
+#' @param nxt Retrieve the next page of Shiny application usage logs 
+#' relative to the provided value. This value corresponds to an internal 
+#' reference within the server and should be sourced from the appropriate 
+#' attribute within the paging object of a previous response.
+#' @param asc_order Defaults to TRUE; Determines if the response records 
+#' should be listed in ascending or descending order within the response. 
+#' Ordering is by the started timestamp field.
+#' 
+#' @return 
+#' A tibble with the following columns:
+#' \itemize{
+#'   \item{\strong{id}}{ID of the audit action}
+#'   \item{\strong{time}}{Timestamp in RFC3339 format when action was taken}
+#'   \item{\strong{user_id}}{User ID of the actor who made the audit action}
+#'   \item{\strong{user_description}}{Description of the actor}
+#'   \item{\strong{action}}{Audit action taken}
+#'   \item{\strong{event_description}}{Description of action}
+#' }
+#' 
+#' @details 
+#' Please see https://docs.rstudio.com/connect/api/#getAuditLogs for more information 
+#' 
+#' @examples 
+#' \dontrun{
+#'   library(connectapi)
+#'   client <- connect()
+#'   
+#'   # get the last 20 audit logs
+#'   get_audit_logs(client, limit = 20, asc_order = FALSE)
+#' }
+#' 
+#' @export
+get_audit_logs <- function(src, limit = 20L, previous = NULL, 
+                           nxt = NULL, asc_order = TRUE){
+  validate_R6_class("Connect", src)
+  
+  # browser()
+  res <- src$audit_logs(
+    limit = limit, 
+    previous = previous, 
+    nxt = nxt, 
+    asc_order = asc_order
+  )
+  
+  res <- page_cursor(src, res, limit = limit)
+  
+  out <- parse_connectapi(res)
+  
+  return(out)
+}
+
+
+
+
+
+
+parse_connectapi <- function(data){
```

Comment:
Maybe worth pulling this into the `utils.R` file? I think this is something that we will want to use generically elsewhere 😄

Reply:
I moved it over! I had originally meant to do that but left it here during development and then forgot!